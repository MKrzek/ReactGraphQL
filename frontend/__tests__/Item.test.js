import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'wwww',
  title: 'hi',
  price: 4003009,
  description: 'sjsjssj',
  image: 'ddd.png',
  largeImage: 'ddddde.png',
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const price = '£50.34';
    expect(price).toMatchSnapshot();
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('renders price tag and title', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    // console.log('wrapper', wrapper.debug());
    // console.log('PriceTag', PriceTag.dive().text());
    expect(PriceTag.children().text()).toBe('£40,030.09');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });
  it('renders buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const ButtonList = wrapper.find('.buttonList');

    expect(ButtonList.children()).toHaveLength(3);
    expect(ButtonList.find('Link')).toBeTruthy();
  });
});
