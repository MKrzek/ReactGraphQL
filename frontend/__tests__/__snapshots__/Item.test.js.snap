// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `"£50.34"`;

exports[`<Item/> renders and matches the snapshot 2`] = `
<ItemStyles__Item>
  <img
    alt="hi"
    src="ddd.png"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "wwww",
          },
        }
      }
    >
      <a>
        hi
      </a>
    </Link>
  </Title>
  <PriceTag>
    £40,030.09
  </PriceTag>
  <p>
    sjsjssj
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "wwww",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="wwww"
    />
    <DeleteItem
      id="wwww"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
